### ISSUE TITLE
Update the Nomad provider for new job features

### ISSUE THREAD

**Issue by shantanugadgil** [terraform-providers/terraform-provider-nomad#46]
### Terraform Version
0.11.8

### Affected Resource(s)
- nomad_job

### Expected Behavior
At any given time, the Nomad provider should have all the features of the latest version of Nomad (v0.8.6 at the time of this bug report)

### Actual Behavior
Terraform's Nomad provider is lagging behind.

### Important Factoids
We (and I assume many other folks) would like to codify their entire usage via Terraform.
The Terraform provider not being able to do many things makes for out-of-band activities like directly having to do `nomad plan`, `nomad run`, etc.

Using the command line is fine for folks who are very familiar with the Nomad options.
All-in-all, it is easier to direct folks to a single "interface" (i.e. Terraform, than to have to learn multiple commands and options)

Thanks and Regards,
Shantanu

**Comment by epetrovich** [comment]
There is a lot of pain in the ss

* nomad_job.blitzops-app-dispatch: 1 error(s) occurred:
* nomad_job.blitzops-app-dispatch: error parsing jobspec: error parsing 'job': 1 error(s) occurred:
* group: 'blitzops-group' -> invalid key: reschedule

**Comment by epetrovich** [comment]
@paddycarver Please!

**Comment by cgbaker** [comment]
Hi all, I am looking into this right now. I will have an update on this ASAP. Thank you for your patience.

**Comment by epetrovich** [comment]
@cgbaker 
I hope you're not in the another time-space dimension )

**Comment by cgbaker** [comment]
I'm in Florida... does that count? 😄

**Comment by mildred** [comment]
Yep, got errors too, at first I thought that the nomad versions I used on my servers was lagging behind, but no, it was here :)

`group: 'main', task: 'main', service (0) -> invalid key: canary_tags`

**Comment by mildred** [comment]
Are there plans to have this provider always in sync (perhaps a new release of this provider on each new nomad release) ?

**Comment by cgbaker** [comment]
We are planning to keep the Terraform provider working with new releases of
Nomad, either by updating the provider on Nomad releases or by modifying
the provider so that it is unversioned/generic.

**Comment by mildred** [comment]
Thanks.

For those interested, I could build the last version of the nomad provider using nomad 0.8.6 with this Dockerfile:

```Dockerfile
FROM golang:1.11 AS terraform-nomad
ARG NOMAD_VERSION=0.8.6
ARG TERRAFORM_NOMAD_VERSION=1.2.0
RUN go get github.com/kardianos/govendor
WORKDIR /go/src/github.com/terraform-providers/terraform-provider-nomad
RUN git clone -b v${TERRAFORM_NOMAD_VERSION} https://github.com/terraform-providers/terraform-provider-nomad.git .
RUN govendor fetch -v github.com/hashicorp/nomad/...@v${NOMAD_VERSION}
RUN govendor fetch -v github.com/hashicorp/golang-lru/...
RUN make build
```

**Comment by cgbaker** [comment]
Yes and yes.

**Comment by shantanugadgil** [comment]
when? and when?

Based on the disturbance in the force (flurry of activity) I sense Consul 1.4.0 or Nomad 0.9.0 (rc1?) is coming and people could be busy! ☺️☺️☺️

**Comment by cgbaker** [comment]
We're going to get the update for 0.8.6 out immediately just by updating
the dependencies, as folks have been doing in this thread. Then we'll work
on un-versioning the client in the provider so that we don't have to keep
working, err, ignoring it. The plan is to have that ship soon after the
0.9.0 release (if not before).

**Comment by ketzacoatl** [comment]
@cgbaker, thanks for the updates! Was there a release made for the provider, updated to work with `0.8.6`?

**Comment by cgbaker** [comment]
Not yet, but soon (hopefully next week).

**Comment by lawliet89** [comment]
It parses and deserialzies the jobspec into some form that the API can consume, is my guess.

**Comment by cgbaker** [comment]
@pashinin , as stated above, the provider is currently "versioned" because of the client that it imports from the Nomad project. The plan is to remove this versioning so that the provider does not need to be constantly updated as new features are added to Nomad.

**Comment by pashinin** [comment]
@cgbaker really waiting for this. Thanks for answers.

**Comment by cgbaker** [comment]
My pleasure. My apologies for the delay on taking care of this.

**Comment by shantanugadgil** [comment]
I am (and have been) waiting for 0.9 GA build.  :) :)

**Comment by ketzacoatl** [comment]
Is there a release coming for this provider to work with nomad `0.8.6`?

**Comment by cgbaker** [comment]
Hopefully next week... There just a few more merge requests I want to pull
in.

**Comment by ketzacoatl** [comment]
@cgbaker would it be reasonable to push a release now with what's ready to go, and then again when there are more PRs merged? Or are there updates that need to block the release?

**Comment by cgbaker** [comment]
It is my only task for today. I'm doing some manual testing right now and then I'll do the release.

**Comment by cgbaker** [comment]
v1.3.0 of the Nomad provider just released, with support for the Nomad v0.8.6 API. Thank you for your patience, I promise you that we will be much faster in adding support in the future. Please let me know if you encounter any problems with this provider.
